### PR TITLE
Remove usage of Node crypto module

### PR DESCRIPTION
**Pull request by RyanZim**
After https://github.com/ExodusMovement/secure-container/pull/4, will need rebase.

### CODE REVIEW

File: lib/crypto.js [hunk 1]
```diff
@@ -15,11 +15,19 @@ exports.boxEncrypt = boxEncrypt;
 exports.boxDecrypt = boxDecrypt;
 exports.sha256 = sha256;
 
-var _crypto = require('crypto');
+var _randombytes = require('randombytes');
```
Comment (mvayngrib) [inline lib/crypto.js:18]: can we remove lib/ from git and publish untranspiled (2 separate questions)?

File: src/crypto.js [hunk 2]
```diff
@@ -14,15 +16,16 @@ export function stretchPassphrase (passphrase: string | Buffer, { salt, n, r, p
   return { key, salt }
 }
 
-export function aesEncrypt (key: Buffer, message: Buffer, opts = { cipher: 'aes-256-gcm', iv: crypto.randomBytes(IV_LEN_BYTES) }) : Object {
-  const cipher = crypto.createCipheriv(opts.cipher, key, opts.iv)
+export function aesEncrypt (key: Buffer, message: Buffer) : Object {
```
Comment (mvayngrib) [inline src/crypto.js:19]: will need to be a major version bump (note to self)

File: src/crypto.js [hunk 3]
```diff
@@ -14,15 +16,16 @@ export function stretchPassphrase (passphrase: string | Buffer, { salt, n, r, p
...
+  const cipher = aes.createCipheriv('aes-256-gcm', key, iv)
   const blob = Buffer.concat([cipher.update(message), cipher.final()])
   const authTag = cipher.getAuthTag()
-  return { authTag, blob, iv: opts.iv }
+  return { authTag, blob, iv }
 }
 
-export function aesDecrypt (key: Buffer, blob: Buffer, { cipher = 'aes-256-gcm', iv, authTag } = {}) : Buffer {
-  const decipher = crypto.createDecipheriv(cipher, key, iv)
+export function aesDecrypt (key: Buffer, blob: Buffer, { iv, authTag } = {}) : Buffer {
```
Comment (mvayngrib) [inline src/crypto.js]: can iv or authTag be undefined?
```suggestion
export function aesDecrypt (key: Buffer, blob: Buffer, { iv, authTag }) : Buffer {
```

**Review by mvayngrib — APPROVED** [review]
utACK

left a few non-blocking comments

File: lib/crypto.js (hunk 1 above)
Comment (farisissa8) [reply lib/crypto.js:18]: Ryan mentioned on a call that the codebase uses flow types so we need to transpile, but maybe we can reduce what transpiling is doing to just strip flowtypes?

File: src/crypto.js [hunk 4]
```diff
@@ -14,15 +16,16 @@ export function stretchPassphrase (passphrase: string | Buffer, { salt, n, r, p
...
+  const decipher = aes.createDecipheriv('aes-256-gcm', key, iv)
```
Comment (farisissa8) [inline src/crypto.js]: Is the change here becaues the module only supports `aes-256-gcm`?

File: lib/crypto.js (hunk 1 above)
Comment (RyanZim) [reply lib/crypto.js:18]: Yeah, this needs to be rethought; just fixing one thing at a time here.

File: src/crypto.js (hunk 2 above)
Comment (RyanZim) [reply src/crypto.js:19]: This is not a documented public method; no need for major.

File: src/crypto.js (hunk 3 above)
Comment (RyanZim) [reply src/crypto.js]: No, good idea.

File: src/crypto.js (hunk 4 above)
Comment (RyanZim) [reply src/crypto.js]: Module supports list here AFAIK: https://github.com/crypto-browserify/browserify-aes/blob/master/modes/list.json

There's just no reason to have an option on an undocumented private method that isn't used.

**Review by farisissa8 — APPROVED** [review]
utACK

**Review by mvayngrib — APPROVED** [review]
utACK

**Pull request merged**